sycl/include/sycl/ext/intel/experimental/esimd/detail/cmrt_if_defs.hpp

```diff
@@ -0,0 +1,163 @@
+//==---------- cmrt_if_defs.hpp - CM-Runtime interface header file ---------==//
```

Comment:
please change the name. esimd_cpu_pi_defs.hpp?

Reply:
Also, this file must not  go into sycl/include/sycl, as this is plugin implementation-related only.
 sycl/plugins/esimd_cpu ?
\+ @smaslov-intel for comments

Reply:
Yeap. This https://github.com/intel/llvm/blob/sycl/sycl/plugins/esimd_cpu/pi_esimd_cpu.hpp seems to be where it needs to go.

Reply:
I'll rename it.

> as this is plugin implementation-related only

No. This file is not plugin implementation-related only. It will be used for kernel compilation too. For example, `_pi_image` is going to be used in `sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp` (https://github.com/intel/llvm/pull/4020/files#diff-53e53ae867a1e31910903247b8d2f9a61331a29a6a38df9afdef5d38bf316fbf)

Reply:
> For example, _pi_image is going to be used in sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp

this is wrong design, unless there are some deep considerations behind this. Kernel device code must not use PI. We need to discuss this. Device code should operate on std sycl types, it may call functions from the CM interface table at most.

Reply:
I'll postpone file renaming (+relocation) after discussion.

Reply:
`_pi_image`,`_pi_buffer`,`_pi_mem`,`_pi_object` definitions are moved to `pi_esimd_cpu.hpp`, and the `cmrt_if_defs.hpp` file is removed.